assigner.py

```diff
@@ -132,23 +133,33 @@ def autocasa(bapi, capi, bcfg, ccfg, dry_run):
                            "using defaults".format(bug.get('assigned_to')))
             delegator_id = casa_status['decidingApprover']['id']
         elif (delegator_id != casa_status['decidingApprover']['id']):
-        ## Set the new assignee if lookup worked
-            res = capi.set_delegator(project_id, delegator_id)
-            logger.info("Setting new assignee/delegator in CASA to {} ({}) for project {}".format(delegator_id,
-                                                                                             bug.get('assigned_to'),
-                                                                                             project_id))
+            ## Set the new assignee if lookup worked
+            if not dry_run:
+                res = capi.set_delegator(project_id, delegator_id)
+                logger.info("Setting new assignee/delegator in CASA to {} ({}) for project {}".format(delegator_id,
+                                                                                               bug.get('assigned_to'),
+                                                                                               project_id))
+            else:
+                logger.info("Would attempt to set assignee/delegator in CASA to {} ({}) for project {}",
+                            "(dry run prevented this)".format(delegator_id, bug.get('assigned_to'), project_id))
         else:
             logger.info("Assignee/delegator in CASA is already correct, no changes made "
                         "{} ({}) for project {})".format(delegator_id, bug.get('assigned_to'), project_id))
 
-        # Update the project status
-        if not dry_run:
-            res = capi.casa_set_status(project_id, delegator_id, bug.get('resolution'))
-            logger.info("CASA API Updated status for project {} to {}: {}".format(project_id, bug.get('resolution'), res))
+        # Update the project status if the bug has been closed in some way
+        if bug.get('status') in ['RESOLVED', 'VERIFIED', 'CLOSED'] :
+            if not dry_run:
+                res = capi.casa_set_status(project_id, delegator_id, bug.get('resolution'))
```

Comment:
On line 150 you support RESOLVED, VERIFIED, CLOSED...  will all of those bug status' also have a relevant bug.get('resolution') response or is that only specific to RESOLVED status bugs.  In other words, have we tried this with VERIFIED and CLOSED status' to make sure it won't stack trace?

Reply:
yes as in the code is not actually modified, just indented due to the new `if` statement
ie, this is what ran in the past, but now it only runs if the bug is in one of the listed statuses. This is because before that, the bug search filtered on these status ahead of time (see https://github.com/mozilla/infosec-risk-management-bugzilla/pull/21/files/d5ac92d71e1ce152a71516c2719df3634d51cd0c#diff-81ed98d8ff392fa5d36f05ccbb76b666L84 - and yes, it was missing `CLOSED` though we never use it in practice, I added it just in case)